src/templates/_image.tpl

```diff
@@ -0,0 +1,22 @@
+{{/*
+Builds a full OCI image reference from the given image object.
+
+Usage:
+  {{ include "papermc-server.image" (dict "image" .Values.container.image) }}
+
+If digest is "none", returns: <registry>/<name>:<tag>
+Otherwise returns: <registry>/<name>@<digest>
+*/}}
+{{ define "papermc-server.image" -}}
+{{ $image := .Values.container.image -}}
+{{ $registry := $image.registry | default "docker.io" -}}
```

Comment:
I feel like it would be a better to deal to directly specify the default values at `values.yaml` level.

This way, we separate the implementation from the default values.